spec/models/group_performance_spec.rb

```diff
@@ -98,4 +98,9 @@
       end
     end
   end
+
+  describe '#groupname' do
+    it { is_expected.to validate_presence_of :groupname }
+    it { is_expected.to validate_uniqueness_of :groupname }
+  end
```

Comment:
Please put it into the `group_spec.rb` file. The above test case is a performance test and the file name reflects that.